low-level/framework/utils/ipmi_client.py

```diff
@@ -238,6 +238,50 @@ def _run_ipmitool_subcommand(self, subcommand, grep_args=None):
 
         return out, error, retcode
 
+    def load_server_fru_list(self):
+        """Get Server FRU list and merge it with server_fru_list,
+        maintained in global config, with which FRU list can be extended
+        for a solution.
+
+        Ex: Supermicro servers not listing disk as FRU,
+        though its most common FRU in servers, and
+        practically it can be replaced easily.
+        So if for a solution, FRU list needs to be extended
+        beyond what server publishes, 'server_fru_list' from global config
+        can be used.
+        Some of the usual FRU examples are:- disk, psu."""
+        cmd = 'fru list'
+        self.fru_list = []
+        output, err, rc = self._run_ipmitool_subcommand(cmd,
+        grep_args="FRU Device Description : ")
+        if rc != 0:
+            logger.error("Failed in fetching FRU info from server."
+                         f"Error:{err}")
+        if output:
+            for line in output.split('\n'):
+                self.fru_list.append(line.split(': ')[1])
+            keywords = ['Pwr Supply', 'power', 'PS', 'PSU']
+            fru_regex = re.compile("|".join(keywords))
+            for fru in self.fru_list:
+                if fru_regex.search(fru):
+                    self.fru_list[self.fru_list.index(fru)] = 'psu'
+            self.fru_list = list(set(self.fru_list))
+        try:
+            default_frus = Conf.get(GLOBAL_CONF,
+                "server_node>server_fru_list>hot_swappable",
```

Comment:
>"server_node>server_fru_list>hot_swappable"
this conf key confirmed by provisioner yet ?
Even if not, yes our code should work, which i can see here. And when provisioner confirms, or changes key, we can update that in upcoming sprint, though make sure we dont miss that

secondly, FRU property exhibition as true/false is fine, what Ujjwal asked was whether its hot-swappable FRU or cold-swappable, so that also needs to reflect now in response. 
So server_fru_list will also have some direct elements under it, though that will be counted as cold-swappable FRUs

> "server_node>server_fru_list"="fan, cpu" ---> dont use this in default config here, though make sure code reads it that way

> "server_node>server_fru_list>hot_swappable" = "disk, psu"

@nitin-seagate please mention is there is problem achieving this

Reply:
> secondly, FRU property exhibition as true/false is fine, what Ujjwal asked was whether its hot-swappable FRU or cold-swappable, so that also needs to reflect now in response.

@cdeshmukh  server_frus fetched from server doesn't give any information about  hot/cold swapped FRU,
to parse this extra information in response json, we have to maintain some kind of mapping in sspl.
And different server gives different format of FRUs, so mapping should be aligned to that.

> server_node>server_fru_list"="fan, cpu" ---> dont use this in default config here, though make sure code reads it that way

are we considering cpu also as fru?

Reply:
yes as discussed in scrum, hot/cold swappable data we are not getting from backend yet, check for storage though. however that data we should source it through a function, which would by default state it as cold_swappable. And in config (solution conf store keys),and along with fru config, hold/cold swappable reference can also stay in config

Reply:
> just keep fan there for server atleast

done. keeping 'fan' in cold_swappable fru list in config